test/test_dataloader.py

```diff
@@ -308,6 +410,52 @@ def test_worker_init_fn(self):
             self.assertEqual(12345, batch[0])
             self.assertEqual(12345, batch[1])
 
+    @unittest.skipIf(IS_WINDOWS, "TODO: need to fix this test case for Windows")
+    def test_interrupt_retry(self):
+        # time.sleep doesn't seem to work on main process when running unittest
+        # for all tests together. Use Process.join as a sleep function.
+        # Case 1: interrupt, check still alove
+        p = ErrorTrackingProcess(target=_test_interrupt_retry)
+        p.start()
+        p.join(JOIN_TIMEOUT)  # give it some time to reach get
+        try:
+            self.assertTrue(p.is_alive())
+            for i in range(3):
+                p.send_signal(signal.SIGUSR1)
+                p.join(0.5)
+                self.assertTrue(p.is_alive())
+            p.join(JOIN_TIMEOUT)
```

Comment:
Why is it supposed to join now? You haven't done anything special before this

Reply:
Good point